docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,395 @@
+content just in `netcore-1.0`
+::: moniker-end
+
+::: moniker range=">netcore-1.0"
+content just in `>netcore-1.0`
+::: moniker-end
+```
+
+> These three level moniker range config should follow these rules:
+> 1. Moniker range is the moniker range expression, and it is composed of one or more comparator sets, joint by `'||'`, `','` is not supported to be a separator.
+> 2. Range requirement:  
+> The range of three level moniker-range config should satisfy a requirement:  
+> `config file` > `YAML header` > `Zone`  
+> Which means the monikerRange specified in last config must be a subset of the range of previous one, and the last config config will be allowed if any of the previous config have not been set.
+> 3. Priority:  
+> `Zone` > `YAML header` > `config file`  
+> The Last config will overwrite the previous one
+
+### 2.2 Link/Xref
+
+In phase 1, content writer is not allowed to reference another file with specific version.
+
+If the referenced file has the same moniker as current page, they will jump to the referenced page with the same moniker. If not, DHS will fallback to the the latest version.
+
+## 3 Output
+
+### 3.1 Output file path
+
+For dynamic, the output path shares the same schema:
+
+`{output-dir}/{local?}/{monikerRangeHash?}/{site-path}`
+
+```
+      locale        monikerRangeHash                      site-path
+      |-^-| |---------------^--------------| |----------------^----------------|
+_site/en-us/01ddf122d54d0f939d1ecf8c6b930ec0/dotnet/api/system.string/index.html
+```
+
+> `?` means optional.
+> When the file have no version, the output path will be `{output-dir}/{local?}/{site-path}`
+
+### 3.2 Output content
+
+- Content for conceptual file
+
+```json
+{
+    ...,
+    "monikers":
+    [
+        "moniker1",
+        ...
+    ]
+    ...,
+}
+```
+
+- Content for Toc
+
+```json
+{
+    ...,
+    {
+        "toc_title": "title",
+        "monikers": [
+            "moniker1",
+            ...
+        ],
+        "children": [...]
+    }
+    ...,
+}
+```
+
+- Content for .manifest.json
+
+    #### Ideal output
+
+    ```json
+    {
+        "files":[
+            {
+                "siteUrl": "/{SitePath}",
+                "outputPath": "{outputPath}",
+                "sourcePath": "{sourcePath}",
+                "monikers": [
+                    "moniker1",
+                    ...
+                ]
+            },
+        ]
+    }
+    ```
+
+    #### Legacy output
+
+    ```json
+    {
+        "groups":[
+            {
+                "id": "{hash of moniker list}",
+                "monikerRange": "{moniker range expression}",
+                "monikers": [
+                    "moniker1",
+                    ...
+                ]
+            },
+            ...
+        ],
+        "files":[
+            {
+                "siteUrl": "/{SitePath}",
+                "outputPath": "{outputPath}",
+                "sourcePath": "{sourcePath}",
+                "monikerRange": "{group id}"
```

Comment:
>"monikerRange": "{group id}" [](start = 16, length = 28)

legacy need "monikers": [ ... ] and "is_moniker_range": true here

Reply:
yes, `is_moniker_range` is needed with value `true`.
but `monikers` is not needed anymore for that the `is_moniker_range` is always true.

---
In reply to: [227289373](https://github.com/dotnet/docfx/pull/3522#discussion_r227289373) [](ancestors = 227289373)